webapp/src/App.js

```diff
@@ -78,53 +84,63 @@ class DecksList extends React.Component {
 
 class DecksListDeck extends React.Component {
   proptypes = {
-    deck: PropTypes.object,
-    api: PropTypes.instanceOf(Api),
+    deck: PropTypes.object.isRequired,
+    deleteDeck: PropTypes.func.isRequired,
   };
 
+  constructor(props) {
+    super(props);
+
+    this.state = { isDeleted: false };
+  }
+
+  handleDeleteButtonClick() {
+    this.setState({ isDeleted: true });
+  }
+
   render() {
+    const { deck, deleteDeck } = this.props;
+    const { isDeleted } = this.state;
+
     return (
-      //TODO: make the fadeout work
-      <Fade in={true} timeout={1000} exit={true}>
+      <Collapse
+        in={!isDeleted}
```

Comment:
This was the revelation. The idea is that we give this `DecksListDeck` an `isDeleted` property in its state, and have it default to false, causing the `in` prop of the `Collapse` component to default to true. When we update the state on click of the delete button, the `DecksListDeck` re-renders itself, and `in` will now be set to false. The transition happens on that rerender.